### PR TITLE
Use ChainerX softmax cross entropy implementation in ChainerX examples

### PR DESCRIPTION
**Pull request by takagi**
Follows #8250 .

https://github.com/takagi/chainer/compare/chainerx-softmax-cross-entropy...takagi:chainerx-example-softmax-cross-entropy

This PR fixes ChainerX examples to use ChainerX softmax cross entropy implementations introduced in #8250 .

### CODE REVIEW

**Comment by hvy** [comment]
#8250 is now merged.

**Comment by takagi** [comment]
Now I've merged the master!

**Comment by hvy** [comment]
Could you fix the style errors?

**Comment by takagi** [comment]
Fixed!

File: chainerx_cc/examples/mnist/train_mnist.cc [hunk 1]
```diff
@@ -145,7 +136,7 @@ void Run(int64_t epochs, int64_t batch_size, int64_t n_hidden, int64_t n_layers,
                 chx::Array x = test_x.At(indices);
                 chx::Array t = test_t.At(indices);
                 chx::Array y = model(x);
-                loss += SoftmaxCrossEntropy(y, t, false);
+                loss += chx::SoftmaxCrossEntropy(y, t);
```
Comment (hvy) [inline chainerx_cc/examples/mnist/train_mnist.cc]: Ditto regarding summation? Also, you might need to include `routines/loss.h`?

File: chainerx_cc/examples/imagenet_py/train_imagenet.py [hunk 2]
```diff
@@ -50,7 +41,7 @@ def evaluate(model, X_test, Y_test, eval_size, batch_size):
             t = Y_test[i:min(i + batch_size, N_test)]
 
             y = model(x)
-            total_loss += compute_loss(y, t) * batch_size
+            total_loss += chx.softmax_cross_entropy(y, t)
```
[the remark this answers is not in the document]
Comment (takagi) [reply chainerx_cc/examples/imagenet_py/train_imagenet.py]: Fixed!

File: chainerx_cc/examples/mnist/train_mnist.cc (hunk 1 above)
Comment (takagi) [reply chainerx_cc/examples/mnist/train_mnist.cc]: Fixed!

**Comment by takagi** [comment]
I missed to apply the change that ChainerX SCE now returns the batch-wise result without reduced. PTAL.

**Comment by hvy** [comment]
Could you merge with the latest master as [it](https://github.com/chainer/chainer/pull/8281) should fix the C++ example?

**Comment by takagi** [comment]
Merged!

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Jenkins, test this please.
